kitty/main.py

```diff
@@ -131,13 +131,25 @@ def main():
         from kitty.client import main
         main(args.replay_commands)
         return
-    if args.single_instance:
+    if args.single_instance or args.remote:
         is_first = single_instance(args.instance_group)
         if not is_first:
```

Comment:
if args.remote and is_first, should raise SystemExit with appropriate error message.